src/layer/Popup.js

```diff
@@ -191,7 +191,8 @@ export var Popup = DivOverlay.extend({
 		if (this.options.closeButton) {
 			var closeButton = this._closeButton = DomUtil.create('a', prefix + '-close-button', container);
 			closeButton.href = '#close';
-			closeButton.innerHTML = '&#215;';
+			// This is actually a button labelled "Close" for accessibility
+			closeButton.innerHTML = '<span role="button" aria-label="Close">&#215;</span>';

```

Comment:
The [HTML validator](https://validator.w3.org/) reports:

> **Error: An element with the attribute `role=button` must not appear as a descendant of the `a` element.**

I suggest setting `role="button"` on the `<a>` element instead.

Reply:
The above suggestion would also be consistent with how it's done for zoom controls:

https://github.com/Leaflet/Leaflet/blob/71f73785c91f50821cee4f590c41b14eeddf394c/src/control/Control.Zoom.js#L91-L92

Reply:
Summary:
```diff
 closeButton.href = '#close';
+closeButton.setAttribute('role', 'button');
+closeButton.setAttribute('aria-label', 'Close popup');
-closeButton.innerHTML = '&#215;';
+closeButton.innerHTML = '<span aria-hidden="true">&#215;</span>';
```


<details><summary>Before & After comparison of computed ARIA properties</summary>
<img src="https://user-images.githubusercontent.com/26493779/89105393-07514f80-d421-11ea-818b-9abea1d96a98.png" alt="popup">
</details>

@nkreer want to move forward with this? 🙏